### DISCUSSION TITLE
How to find the Excitation magnitudes and phases of HFSS design

### DISCUSSION THREAD

**Discussion by jtom95** [ansys/pyaedt#3471] [Q&A]
Hi all, 

I have not beee able to obtain the list of source magnitudes and phases. Indeed, I am able to edit the sources with the following: 

```

solution_module = hfss.get_module("Solutions")
    for M, d_name in zip(moments, dipole_names):
        print(f"{np.real(M)}Vm + {np.imag(M)}i Vm")
        print(d_name)
        solution_module.EditSources(
            [
                "FieldType:="       , "TotalFields",
                "IncludePortPostProcessing:=", False,
                "SpecifySystemPower:="  , False
            ])
       
        magnitude = np.abs(M)
        phase = np.angle(M, deg=True)
       
        solution_module.EditSources(
            [
                "Name:="        , d_name,
                "Magnitude:="       , f"{magnitude}{units}",
                "Phase:="       , f"{phase}deg"
            ]
        )

``` 
However, if I use the same module to try and find the sources, i.e solution_module.GetAllSourceMagnitudes() or solution_module.GetAllSourcePhases(), I get a list of 1s and 0s. While the real magnitudes and phases are different. In my particular example here is the list: 

![image](https://github.com/ansys/pyaedt/assets/66060633/05fe01cc-e7c9-49d7-a4ae-8e4c59a12b31)

I hope someone can help.
Thank you. 
Tomas

**Comment by Samuelopez-ansys** [comment]
Hi @jamestom-cpu ,

This is an AEDT API bug. Please next time if you find something like this, contact Ansys support and they will submit a bug to the AEDT development team.

Then regarding this discussion, there is  no way to get the edit sources with currents. I have tried with lumped ports and waveports and it worked.

**Reply by jtom95** [reply]
Thanks for the reply @Samuelopez-ansys. Do you think there is a way using the original iron python automation?
I first wrote about this issue on the learning forum and they directed me here :)

**Reply by Samuelopez-ansys** [reply]
@jamestom-cpu There is no way, I have submitted a defect internally, and it will be fixed in 2024R2.
